pkg/http/routes.go

```diff
@@ -30,3 +30,27 @@ func MetricRoutes(mux *http.ServeMux) *http.ServeMux {
 	mux.Handle("/metrics", promhttp.Handler())
 	return mux
 }
+
+// ReloadRoutes adds the reload endpoint to a mux.
+func ReloadRoutes(mux *http.ServeMux) (*http.ServeMux, <-chan func(error)) {
```

Comment:
Let's simplify the ReloadRoutes func to `func ReloadRoutes(mux *http.ServeMux, reloader func() error) (*http.ServeMux)` and implement the channel mangling out of band, i.e. in `main`:

```
reloader := func() error {
  errChan := make(chan error)
  reloadChan <- func(err error) {
    errChan <- err
  }
  return <-errChan
}
```

and then the ReloadRoutes func body becomes as simple as:
```
mux.HandleFunc("/-/reload", func(w http.ResponseWriter, req *http.Request) {
		if req.Method != http.MethodPost {
			w.WriteHeader(http.StatusMethodNotAllowed)
			return
		}
		err := reloader()
		if err != nil {
			w.WriteHeader(http.StatusInternalServerError)
			return
		}
	})
```

what do you think?

Reply:
Sounds great to me :) . This is very nice. It shuffles the complexity around, but keeps the handler quite clean.

Reply:
You know what, I don't think we actually really need to put the reload handler logic in a runGroup. If we pass the `Reconfigure` func directly to the handler, we can simplify things even more.

Reply:
@squat even better! 👍 the less concurrency, the better!